### PR TITLE
DROOLS-3381: [DMN Designer] Do not add Header to nested ContextGrid

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-3381

This PR adds support for columns with zero headers.

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/556
- https://github.com/kiegroup/kie-wb-common/pull/2316

**Comment by manstis** [comment]
@karreiro @jomarko For your review please.

**Comment by kiereleaseuser2** [comment]
Build finished. 5033 tests run, 11 skipped, 0 failed.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by kiereleaseuser2** [comment]
Build finished. 5033 tests run, 11 skipped, 0 failed.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java [hunk 1]
```diff
@@ -298,7 +298,7 @@ public int getHeaderRowCount() {
 
     @Override
     public void setHeaderRowCount(final int headerRowCount) {
-        if (!(headerRowCount > 0)) {
+        if (!(headerRowCount >= 0)) {
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java]: I slightly think we could remove this `setHeaderRowCount` method completely as the `getHeaderRowCount` is kind of independent and uses `headerRowCount` private field just as some default return value for case whn no column are specified or they have no header metadata , is this correct?

Probably we won't risk regressions so keep this method, however please update javadoc in `GridData` interface.

**Comment by manstis** [comment]
@karreiro Could you please review at your convenience?

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRendererTest.java [hunk 2]
```diff
@@ -98,6 +104,22 @@ public void setup() {
         when(rc.isSelectionLayer()).thenReturn(isSelectionLayer());
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
```
[the remark this answers is not in the document]
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRendererTest.java]: Done.

**Comment by kiereleaseuser2** [comment]
Build finished. 5038 tests run, 11 skipped, 0 failed.

**Comment by manstis** [comment]
@ederign Could you please merge this and the related https://github.com/kiegroup/kie-wb-common/pull/2316

**Comment by manstis** [comment]
@ederign Pleeeease merge at your convenience :-)

**Pull request merged**